tests/gstreamer_join/unittest_join.cc

```diff
@@ -0,0 +1,241 @@
+
+  buf_0 = gst_buffer_new ();
+  mem = gst_allocator_alloc (NULL, 192, NULL);
+  gst_memory_map (mem, &info, GST_MAP_WRITE);
+  memcpy (info.data, test_frames[0], 192);
+  gst_memory_unmap (mem, &info);
+  gst_buffer_append_memory (buf_0, mem);
+  buf_3 = gst_buffer_copy (buf_0);
+
+  buf_1 = gst_buffer_new ();
+  mem = gst_allocator_alloc (NULL, 192, NULL);
+  gst_memory_map (mem, &info, GST_MAP_WRITE);
+  memcpy (info.data, test_frames[1], 192);
+  gst_memory_unmap (mem, &info);
+  gst_buffer_append_memory (buf_1, mem);
+  buf_4 = gst_buffer_copy (buf_1);
+
+  data_received = 0;
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (100000);
+
+  idx = 0;
+  EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc_handle_0), buf_0), GST_FLOW_OK);
+  g_usleep (100000);
+
+  idx = 1;
+  EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc_handle_1), buf_1), GST_FLOW_OK);
+  g_usleep (100000);
+
+  EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc_handle_1), buf_4), GST_FLOW_OK);
+  g_usleep (100000);
+
+  g_object_get (join_handle, "active-pad", &active_pad, NULL);
+  EXPECT_NE (nullptr, active_pad);
+  active_name = gst_pad_get_name (active_pad);
+  EXPECT_STREQ ("sink_1", active_name);
+  gst_object_unref (active_pad);
+  g_free (active_name);
+
+  idx = 0;
+  EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc_handle_0), buf_3), GST_FLOW_OK);
+  g_usleep (100000);
+
+  g_object_get (join_handle, "active-pad", &active_pad, NULL);
+  EXPECT_NE (nullptr, active_pad);
+  active_name = gst_pad_get_name (active_pad);
+  EXPECT_STREQ ("sink_0", active_name);
+  gst_object_unref (active_pad);
+  g_free (active_name);
+
+  g_object_get (join_handle, "n-pads", &n_pads, NULL);
+  EXPECT_EQ (2, n_pads);
+
+  gst_object_unref (sink_handle);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (100000);
+  EXPECT_EQ (4, data_received);
+
+  gst_object_unref (appsrc_handle_0);
+  gst_object_unref (appsrc_handle_1);
+  gst_object_unref (join_handle);
+  gst_object_unref (pipeline);
+  g_free (str_pipeline);
+}
+
+/**
+ * @brief Test get property with invalid parameter
+ */
+TEST (join, prop_0_n)
+{
+  GstElement *join_handle;
+  gchar * str_val = NULL;
+
+  gchar *str_pipeline = g_strdup
+      ("appsrc name=appsrc_0 ! other/tensor,dimension=(string)3:4:2:2,type=(string)int32,framerate=(fraction)0/1 ! join.sink_0 "
+       "appsrc name=appsrc_1 ! other/tensor,dimension=(string)3:4:2:2,type=(string)int32,framerate=(fraction)0/1 ! join.sink_1 "
+       "join name=join ! other/tensor,dimension=(string)3:4:2:2, type=(string)int32, framerate=(fraction)0/1 ! "
+       "tensor_sink name=sinkx async=false");
+
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  EXPECT_NE (pipeline, nullptr);
+
+  join_handle = gst_bin_get_by_name (GST_BIN (pipeline), "join");
+  EXPECT_NE (join_handle, nullptr);
```

Comment:
When considering this unittest's purpose, I think the checking for `pipeline` & `join_handle` should be `assert`, not `expect`

Reply:
Thanks. Updated.